src/overlays/actors/ovl_Obj_Warpstone/z_obj_warpstone.c

```diff
@@ -22,39 +30,156 @@ const ActorInit Obj_Warpstone_InitVars = {
+        Math_StepToF(&this->dyna.actor.velocity.x, 0.01f, 0.001f);
+        Math_StepToS(&this->dyna.actor.home.rot.x, 0xFF, 0x12);
+    } else {
+        Math_StepToF(&this->dyna.actor.velocity.x, 20.0f, 0.01f);
+        if (this->dyna.actor.velocity.x > 0.2f) {
+            this->modelIndex = SEK_MODEL_OPENED;
+            Math_StepToS(&this->dyna.actor.home.rot.x, 0, 0x14);
+        }
+    }
+    return true;
+}
+
+s32 ObjWarpstone_OpenedIdle(ObjWarpstone* this, GlobalContext* globalCtx) {
+    /*You can save your progress and quit here.*/
+    this->dyna.actor.textId = 0xC01;
+    return false;
+}
+
+void ObjWarpstone_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjWarpstone* this = THIS;
+    s32 pad;
+
+    if (this->isTalking) {
+        if (func_800B867C(&this->dyna.actor, globalCtx) != 0) {
+            this->isTalking = false;
+        } else if ((func_80152498(&globalCtx->msgCtx) == 4) && (func_80147624(globalCtx))) {
+            if (globalCtx->msgCtx.choiceIndex != 0) {
+                func_8019F208();
+                globalCtx->msgCtx.unk11F22 = 0x4D;
+                globalCtx->msgCtx.unk120D6 = 0;
+                globalCtx->msgCtx.unk120D4 = 0;
+                gSaveContext.owlSaveLocation = GET_OWL_ID(this);
+            } else {
+                func_801477B4(globalCtx);
+            }
+        }
+    } else if (func_800B84D0(&this->dyna.actor, globalCtx)) {
+        this->isTalking = true;
+    } else if (!this->actionFunc(this, globalCtx)) {
+        func_800B863C(&this->dyna.actor, globalCtx);
+    }
+    Collider_ResetCylinderAC(globalCtx, &this->collider.base);
+    Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
+
+void ObjWarpstone_Draw(Actor* thisx, GlobalContext* globalCtx2) {
+    ObjWarpstone* this = THIS;
+    GlobalContext* globalCtx = globalCtx2;
+
+    func_800BDFC0(globalCtx, sOwlStatueDLs[this->modelIndex]);
+    if (this->dyna.actor.home.rot.x != 0) {
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+        func_8012C2DC(globalCtx->state.gfxCtx);
+        SysMatrix_InsertTranslation(this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y + 34.0f,
+                                    this->dyna.actor.world.pos.z, 0);
+        SysMatrix_InsertMatrix(&globalCtx->mf_187FC, 1);
+        SysMatrix_InsertTranslation(0.0f, 0.0f, 30.0f, 1);
+        Matrix_Scale(this->dyna.actor.velocity.x, this->dyna.actor.velocity.x, this->dyna.actor.velocity.x, 1);
+        SysMatrix_StatePush();
+        gDPPipeSync(POLY_XLU_DISP++);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0x80, 0x80, 0xFF, 0xFF, 0xC8, this->dyna.actor.home.rot.x);
+        gDPSetEnvColor(POLY_XLU_DISP++, 0x64, 0xC8, 0x00, 0xFF);
+        SysMatrix_InsertZRotation_f((((globalCtx->gameplayFrames * 1500) & 0xFFFF) * M_PI) / 32768.0f, 1);
```

Comment:
Would something like 

```SysMatrix_InsertZRotation_f(RADF_TO_BINANG(globalCtx->gameplayFrames * 1500), MTXMODE_APPLY);```

work? I think this is a macro regardless

Reply:
Nope, that doesn't match sadly.

Reply:
The `0xFFFF` was dropped and that isn't included in the macro:
`RADF_TO_BINANG((globalCtx->gameplayFrames * 1500) & 0xFFFF) should match (plus see my comment on `% 65536`

Reply:
Somehow that doesn't match either. It removes like a dozen or so lines of assembly from the function.

Reply:
% 65536 is working though. Would it be better to have that as hex? `% 0x10000`? 65536 is a "nothing special" number in decimal, but in hex it's a big deal (adds a new digit).

Reply:
s16 Angles should be in hex unless there's a really good reason not to (floats, some weird matching shenanigans).

Reply:
So there's two occurrences of this, and one I can replace with `% 0x10000` without any issues, but the other will only match with `& 0xFFFF`. Should I just keep both as `& 0xFFFF` for consistency at that point?

Reply:
If a `u16` cast works, go with that, otherwise stick with the `& 0xFFFF`s IMO. It's pretty commont to see that in their code anyway.

Reply:
u16 didn't work so i stuck with & 0xFFFF.